### ISSUE TITLE
Creating job with an undefined priority causes 

### ISSUE THREAD

**Issue by TheGrandmother** [Automattic/kue#1061]
Dsing the `priority` method with an undefined value causes when creating a job causes `node_redis` to complain with:
```
node_redis: Deprecated: The HSET command contains a "undefined" argument.
This is converted to a "undefined" string now and will return an error from v.3.0 on.
Please handle this in your code to make sure everything works as you intended it to.
```

Would it not be preferable to have Kue throw an exception  if the priority is undefined?
It took me some time to find the cause of this problem in my code :p

**Comment by behrad** [comment]
I'd rather like kue to set a default priority, and I expected so!
Can you confirm and create a PR if not?

**Comment by TheGrandmother** [comment]
It does not set the default priority when `job.priority` is used with either `undefined` as argument or anything that is not in the priority map or a function.

I can fix it.
To what branch do you want the pull?

**Comment by TheGrandmother** [comment]
@behrad 

Why is there a `'critical'` priority in:
```
var priorities = exports.priorities = {
  low: 10, normal: 0, medium: -5, high: -10, critical: -15
};
``` 
whilst the docs for the `Job.prototype.priority` does not mention it
and states that the priority should be between -10 and 10?

Is `'critical'` priority reserved for the internal implementation?
Should the user be able to set it?

**Comment by behrad** [comment]
> Is 'critical' priority reserved for the internal implementation?

No



> whilst the docs for the Job.prototype.priority does not mention it
> and states that the priority should be between -10 and 10?

Documentation Issue, user can use and set to it :)

**Comment by behrad** [comment]
I'd expect all jobs priority to be normal if not set by user :)

**Comment by TheGrandmother** [comment]
@behrad 
I agree that the default should `normal` what i meant to say was that if the argument to priority is malformed an error/warning should be given and not silently set to default.
